### DISCUSSION TITLE
Analog Inference 1T1R? 1R?

### DISCUSSION THREAD

**Discussion by HeatPhoenix** [IBM/aihwkit#659] [General] [answered]
What are the assumptions made in the IBM AI HW Kit with regards to access transistors? Are they implemented, are they assumed to be present (individually accessible devices)? The only reference to transistors I could find in the project was in: https://github.com/IBM/aihwkit/blob/69e524fbc2c5462cbe1a12ee1662879c001c30c0/notebooks/analog_fusion.ipynb#L22 wherein it is discussed in the context of the fusion chip (which is also the basis of the PCM data in the toolkit, if I understand it correctly)?

**Comment by maljoras-sony** [comment, marked as answer]
Hi @HeatPhoenix,
many thanks for the question. Typically it is assumed *not* assumed that individual devices are accessible directly and individually. If one wants to readout devices, one has to perform an MVM, e.g. with one-hot input, which is then subject to noise etc. That said, for debugging purposes it is of course possible to readout the conductances exactly. For instance, `layer.get_weight(apply_weights_scaling=False)` gets the part of the weights that are in (normalized) conductances (note that there are possibly digital output scales part of the overall weight) in perfect manner (without any readout noise). However, in case of `layer.get_weight(apply_weights_scaling=False, realistic=True)` multiple MVMs are performed to read out the weights in a realistic manner (without direct access to individual weights). Similarly, e.g. drift compensation is performed by reading out the weights in a realistic manner (without direct access). 

Hope that answers your question.

**Reply by HeatPhoenix** [reply]
Answers the question perfectly. Thank you.
